procat/settings/settings.py

```diff
@@ -179,6 +173,35 @@
 OIDC_RP_SCOPES = os.environ.get("OIDC_RP_SCOPES")
 OIDC_RP_SIGN_ALGO = os.environ.get("OIDC_RP_SIGN_ALGO", "RS256")
 
+
+def is_non_empty(value: str | None) -> bool:
+    """Return True if the given environment variable is set and non-empty."""
+    return value is not None and str(value).strip() != ""
+
+
+USE_OIDC = all(
+    [
+        is_non_empty(OIDC_RP_CLIENT_ID),
+        is_non_empty(OIDC_RP_CLIENT_SECRET),
+        is_non_empty(OIDC_OP_AUTHORIZATION_ENDPOINT),
+        is_non_empty(OIDC_OP_TOKEN_ENDPOINT),
+        is_non_empty(OIDC_OP_USER_ENDPOINT),
+        is_non_empty(OIDC_OP_JWKS_ENDPOINT),
+    ]
+)
+
+if USE_OIDC:
+    AUTHENTICATION_BACKENDS = [
+        "django.contrib.auth.backends.ModelBackend",
+        "main.oidc.ICLOIDCAuthenticationBackend",
+    ]
+    LOGIN_URL = "/oidc/authenticate/"
+else:
+    AUTHENTICATION_BACKENDS = [
+        "django.contrib.auth.backends.ModelBackend",
+    ]
+    LOGIN_URL = "/register/"
```

Comment:
This should be:

```suggestion
    LOGIN_URL = "/auth/login/"
```
To log users in if they are not. `register` is used to register new users.